### PR TITLE
Add a hook to ensure the client's version is updated before pushing a tag/release

### PR DESCRIPTION
**Pull request by MihaiBojin**
## Description

- Versioned the user agent string

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/mongodbatlas_test.go [hunk 1]
```diff
@@ -93,8 +93,8 @@ func testClientDefaultBaseURL(t *testing.T, c *Client) {
 }
 
 func testClientDefaultUserAgent(t *testing.T, c *Client) {
-	if c.UserAgent != userAgent {
-		t.Errorf("NewClient UserAgent = %v, expected %v", c.UserAgent, userAgent)
+	if c.UserAgent != DefaultUserAgent() {
```
Comment (MihaiBojin) [inline mongodbatlas/mongodbatlas_test.go:96]: updated the test suite to expect the new UA

File: mongodbatlas/version.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const (
+	// UserAgent the default user agent string reported by this client
+	UserAgent = "go-mongodbatlas"
+	// ClientVersion the version of the current API client
+	ClientVersion = "0.4.0" // Should be set to the next version planned to be released
+)
+
+// DefaultUserAgent constructs the default user agent
+func DefaultUserAgent() string {
```
Comment (MihaiBojin) [inline mongodbatlas/version.go:13]: helper function to construct the default UA string

File: mongodbatlas/version.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package mongodbatlas
```
Comment (gssbzn) [inline mongodbatlas/version.go:1]: [non-blocking] I don't feel moving this to a version file adds any value vs the mongodbatlas. If we were to have a package `version` then moving _only_ the `ClientVersion` there would make more sense

File: mongodbatlas/version.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+import "fmt"
+
+const (
+	// UserAgent the default user agent string reported by this client
+	UserAgent = "go-mongodbatlas"
```
Comment (gssbzn) [inline mongodbatlas/version.go:7]: I think we should not export this
```suggestion
	userAgent = "go-mongodbatlas"
```

File: mongodbatlas/version.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	// ClientVersion the version of the current API client
+	ClientVersion = "0.4.0" // Should be set to the next version planned to be released
```
Comment (gssbzn) [inline mongodbatlas/version.go:9]: it's more common to just say `Version`
```suggestion
	// Version the version of the current API client
	Version = "0.4.0" // Should be set to the next version planned to be released
```

File: mongodbatlas/version.go (hunk 2 above)
Comment (gssbzn) [inline mongodbatlas/version.go:13]: [nit] similar to the const I don't see a value on exporting this or even why we have 
`UserAgent` as a const and `DefaultUserAgent` as a func
we should just offer one, maybe 

```go
const libraryName = "go-mongodbatlas"
const Version = "0.4"
var userAgent = fmt.Sprintf("%s/%s (%s;%s)", libraryName, Version, runtime.GOOS, runtime.GOARCH)
```

File: mongodbatlas/version.go (hunk 5 above)
Comment (MihaiBojin) [reply mongodbatlas/version.go:9]: version is very generic though; hard to search for

File: mongodbatlas/version.go (hunk 3 above)
Comment (MihaiBojin) [reply mongodbatlas/version.go]: yeah fair enough, updated

File: mongodbatlas/version.go (hunk 5 above)
Comment (MihaiBojin) [reply mongodbatlas/version.go]: still, updated since it's more go idiomatic

File: mongodbatlas/version.go (hunk 2 above)
Comment (MihaiBojin) [reply mongodbatlas/version.go]: nice!

**Comment by gssbzn** [comment]
in mcli we have a [RELEASING.md](https://github.com/mongodb/mongocli/blob/master/RELEASING.md) we also have a little bash script to automate some of the process, we could look if making a bash script to update the version and tag the release would make sense here

**Comment by themantissa** [comment]
@gssbzn this is an excellent example, thank you!  Sounds like we'll want similar for this but perhaps figure out communication channels between teams for timing, etc.

File: .github/workflows/release-check.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    name: Release sanity tests
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Checking code health
+        run: make tools lint test
+
+      - name: Checking the version
+        run: make check-version
```
Comment (MihaiBojin) [inline .github/workflows/release-check.yml:18]: This workflow will ensure that we get notified when a tag was pushed and the version was not updated.

File: GNUmakefile [hunk 7]
```diff
@@ -13,26 +13,36 @@ link-git-hooks:
 	find .git/hooks -type l -exec rm {} \;
 	find githooks -type f -exec ln -sf ../../{} .git/hooks/ \;
 
+.PHONY: build
```
Comment (MihaiBojin) [inline GNUmakefile:16]: updated for consistency

File: README.md [hunk 8]
```diff
@@ -67,6 +67,9 @@ Each version of the client is tagged and the version is updated accordingly.
 
 To see the list of past versions, run `git tag`.
 
+To release a new version, first ensure that `[Version](./mongodbatlas/mongodbatlas.go)' is updated
```
Comment (gssbzn) [inline README.md:70]: should 
>`[Version](./mongodbatlas/mongodbatlas.go)'
be
`[Version](./mongodbatlas/mongodbatlas.go)` 
also does the linking works like this? sometimes for file you need to add blob in the path

File: scripts/check-version.sh [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    echo "ERROR: Mismatch between the release version ('$tag') and '$VERSION_FILE'"
+    echo "$VERSION_FILE:"
+    echo "---------"
+    grep "Version = " "$VERSION_FILE"
+    echo "---------"
+    echo
+    exit 1
+fi
+echo "OK."
+echo
```
Comment (gssbzn) [inline scripts/check-version.sh:31]: [super-nit] missing new line

File: README.md (hunk 8 above)
Comment (MihaiBojin) [reply README.md]: nice catch on the backticks!
Yup, linking like that (against a file or dir, relative to the current file) works due to GitHub Markdown syntax; e.g.: https://github.com/mongodb/go-client-mongodb-atlas/blob/ed14165966a89e84a67e79ad7aeabedbab2734b9/README.md

**Pull request merged**